mlrun/runtimes/function.py

```diff
@@ -246,97 +248,83 @@ def serving(
 
         return self
 
-    def deploy(self, dashboard="", project="", tag="", kind=None, env: dict = None):
-        def get_fullname(config, name, project, tag):
-            if project:
-                name = "{}-{}".format(project, name)
-            if tag:
-                name = "{}-{}".format(name, tag)
-            update_in(config, "metadata.name", name)
-            return name
-
-        self.set_config("metadata.labels.mlrun/class", self.kind)
-        env_dict = {get_item_name(v): get_item_name(v, "value") for v in self.spec.env}
-        if env:
-            for name, value in env.items():
-                env_dict[name] = value
-        spec = nuclio.ConfigSpec(env=env_dict, config=self.spec.config)
-        spec.cmd = self.spec.build.commands or []
-        project = project or self.metadata.project or "default"
-        handler = self.spec.function_handler
-
-        # In Nuclio 1.6.0 default serviceType changed to "ClusterIP", make sure we're using NodePort
-        spec.set_config("spec.serviceType", "NodePort")
-        if self.spec.readiness_timeout:
-            spec.set_config("spec.readinessTimeoutSeconds", self.spec.readiness_timeout)
-        if self.spec.resources:
-            spec.set_config("spec.resources", self.spec.resources)
-        if self.spec.no_cache:
-            spec.set_config("spec.build.noCache", True)
-        if self.spec.replicas:
-            spec.set_config("spec.minReplicas", self.spec.replicas)
-            spec.set_config("spec.maxReplicas", self.spec.replicas)
+    def deploy(
+        self, dashboard="", project="", tag="", verbose=False,
+    ):
+        verbose = verbose or self.verbose
+        if project:
+            self.metadata.project = project
+        if tag:
+            self.metadata.project = tag
+        state = ""
+        last_time = 1
+
+        if not dashboard:
+            db = self._get_db()
+            logger.info("starting remote function deploy")
+            data = db.remote_builder(self, False)
+            self.status = data["data"].get("status", None)
+            # ready = data.get("ready", False)
+
+            while state not in {"ready", "error"}:
+                sleep(1)
+                try:
+                    text, last_time = db.get_builder_status(
+                        self, last_time=last_time, verbose=verbose
+                    )
+                except RunDBError:
+                    raise ValueError("function or deploy process not found")
+                state = self.status.state
+                if text and state != "error":
+                    print(text)
```

Comment:
Why not printing when `state != "error"` ?

Reply:
@Hedingber in error state it prints it w error log and in the raise msg, so already 2 times, this will be the 3rd